### ISSUE TITLE
Investigate the possiblity and method to replace GLOG with spdlog

### ISSUE THREAD

**Issue by PragmaTwice** [apache/kvrocks#2267]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

spdlog (https://github.com/gabime/spdlog) is a modern logging library based on fmt library (or `std::format`).

It's better to use spdlog instead of glog because we extensively utilize the fmt library.

Note: we need to specify `SPDLOG_FMT_EXTERNAL` since we need to keep there's only one fmt library in our deps.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by YaroslavStepanchuk** [comment]
@PragmaTwice Good idea. However, we use this function from the glog library https://github.com/apache/kvrocks/blob/unstable/src/cli/signal_util.h#L54. 

I was looking for a way to get rid of it because in the glog-0.7.0 it's no more public. 

N.B. I tried to use the libunwind but failed to add it to the project. However, it's a different story :)

**Comment by 133tosakarin** [comment]
hello, I'm a beginer, Can I choose this issue ?

**Comment by git-hulk** [comment]
@133tosakarin Perhaps you can try #2266 first.

**Comment by letian-jiang** [comment]
@git-hulk @PragmaTwice Could you assign this one to me?

BTW, is there any expected performance improvement after replacing GLOG with spdlog?

**Comment by PragmaTwice** [comment]
> BTW, is there any expected performance improvement after replacing GLOG with spdlog?

I think it's good if there's no noticeable performance degradation.
